internal/response_asserter/response_asserter.go

```diff
@@ -72,12 +88,47 @@ func (a ResponseAsserter[ResponseType]) DecodeAndAssertSingleFields(responsePayl
 	return actualResponse, nil
 }
 
-func (a ResponseAsserter[ResponseType]) DecodeAndAssert(responsePayload []byte) (ResponseType, error) {
-	actualResponse, err := a.DecodeAndAssertSingleFields(responsePayload)
+func (a ResponseAsserter[ResponseType]) DecodeAndAssert(response kafka_client.Response) (ResponseType, error) {
+	actualResponse, err := a.DecodeAndAssertSingleFields(response)
 
 	if err != nil {
 		return actualResponse, err
 	}
 
 	return actualResponse, a.Assertion.AssertAcrossFields(actualResponse, a.Logger)
 }
+
+func (a ResponseAsserter[ResponseType]) assertMessageLength(response kafka_client.Response) error {
+	if len(response.RawBytes) < 4 {
+		return fmt.Errorf("Response is not long enough to accomodate message size.")
```

Comment:
It might not be clear what "message size" is here. Also, note that this doesn't have the byte diff stuff so a user isn't going to see the exact bytes they sent either. 

Can we also have a fixture for this? Need to actually see this to know what it actually looks like